src/main/java/org/pac4j/dropwizard/Pac4jConfiguration.java

```diff
@@ -1,7 +1,172 @@
 package org.pac4j.dropwizard;
 
-import io.dropwizard.Configuration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-public interface Pac4jConfiguration<T extends Configuration> {
-    Pac4jFactory getPac4jFactory(T configuration);
+import org.pac4j.core.authorization.authorizer.Authorizer;
+import org.pac4j.core.authorization.generator.AuthorizationGenerator;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.matching.Matcher;
+import org.pac4j.jax.rs.filter.SecurityFilter;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An interface defining getters for parameters required to configure a {@link SecurityFilter}.
+ */
+public class Pac4jConfiguration {
```

Comment:
It's exactly what I want: that the user could continue configuring pac4j after its config has been populated by Pac4jBundle :)

I see your point theoretically for the non-need to modify it, but in practice it is not always true!
For example, with pac4j, I would like to be able to setup one of the authenticator with a connection to the database: how can I do that at another time than during the application.run method where I actually instantiate such a connection (via its factory's build method for example).
If you see another idiomatic way of doing that, I'm all for it, but even in the past when I only needed to access a Bundle's configuration, it was a problem because the configuration was built by a factory, so once created, the instance is not accessible anymore to anybody except the bundle! No?